### ISSUE TITLE
Build URL support for project searches

### ISSUE THREAD

**Issue by alanmoo** [mozilla/science.mozilla.org#328]
So that search result sets can be passed back and forth via links

**Comment by hannahkane** [comment]
This is a prereq for building out #509

**Comment by akshatnitd** [comment]
@alanmoo This can be done via 3 ways - 

1. Passing the search results as the **state** key in the **to** object of the `<Link>` tag. The link to this approach can be found [here](https://github.com/ReactTraining/react-router/blob/v2.8.1/docs/API.md#to ).

2. Use `localStorage` to store the search results and then while rendering the other component, get value back from it.

3. This approach first needs **_redux_** to be implemented. We can pass the **id** of the search results as a **param** in the `<Link>` tag and then get the value from the redux store if available, or make an API request to fetch the result.

**Comment by alanmoo** [comment]
I think you're overthinking it. I haven't looked at the code recently, but see if this makes sense:

From the project page view you should be able to access the url query params, and pass that into the `get` request (the API already supports searching by tags). That way, anyone hitting a url like `https://science.mozilla.org/projects?tag=Javascript` will get a list of projects on the existing project page. We may want to add some functionality to auto scroll to the list of projects if there's a query param.

(I should also note the expectation for this ticket is that the query params listed at https://api-mozillascience-production.herokuapp.com/projects/ should all work, except for `expand`)

**Comment by akshatnitd** [comment]
@alanmoo Yes, that sounds good. I thought it the other way. Will be sending the PR soon :+1:
